platform/util/src/com/intellij/openapi/util/Key.java

```diff
@@ -106,6 +106,17 @@ public void set(@Nullable Map<Key, Object> holder, T value) {
     return (Key<T>)allKeys.get(index);
   }
 
+  @NotNull
+  public static Key[] getKeysByIndices(int[] indexes) {
```

Comment:
Index of a key is an implementation detail and shouldn't be exposed to open API. It's bad that we already have public 'getKeyByIndex' method, I would prefer to make it package-local. But anyway let's not emphasize this problem by adding implementation-specific utility method to API.